### PR TITLE
BUILD-469: add webhook main entrypoint and dockerfile

### PR DESCRIPTION
**Pull request by jkhelil**
referes to https://issues.redhat.com/browse/BUILD-469

### CODE REVIEW

**Comment by gabemontero** [comment]
/retitle BUILD-469: add webhook main entrypoint and dockerfile

**Comment by gabemontero** [comment]
so yeah presumably we need https://github.com/openshift/release/pull/27887 to merge for the images job to start passing here with the moving of the Dockerfile locations

File: images/Dockerfile
Comment (gabemontero) [inline images/Dockerfile:3]: revert this change too

**Review by gabemontero — CHANGES_REQUESTED** [review]

File: cmd/csidriver/main_test.go
Comment (gabemontero) [inline cmd/csidriver/main_test.go:62]: not sure why this is necessary ... but the e2e's should sort it out once we have fixed the image building stuff between here and the updated openshift/release PR

File: images/Dockerfile.webhook
Comment (jkhelil) [reply images/Dockerfile.webhook]: done

**Review by gabemontero — CHANGES_REQUESTED** [review]
So @jkhelil I'll give you 2 choices based on the guidance we got from Ben in your openshift/release PR

1) just move the Dockerfiles out of the image sub directory back into the top level directory, and delete the images subdirectory

2) get what you were intending with 3 PRs
   a) this PR creates Dockerfiles in both the top level dir and the images dir ... once https://github.com/openshift/release/pull/27951 we should get clean runs, we can merge this
   b) after this merges, you create a new openshift/release PR that points to the dockerfiles in the images subdir
   c) after b) merges, create a new PR here that deletes the dockerfiles in the top level dir

I'll let you decide what you want to do

**Comment by jkhelil** [comment]
> So @jkhelil I'll give you 2 choices based on the guidance we got from Ben in your openshift/release PR
> 
> 1. just move the Dockerfiles out of the image sub directory back into the top level directory, and delete the images subdirectory
> 2. get what you were intending with 3 PRs
>    a) this PR creates Dockerfiles in both the top level dir and the images dir ... once [openshift/release#27951](https://github.com/openshift/release/pull/27951) we should get clean runs, we can merge this
>    b) after this merges, you create a new openshift/release PR that points to the dockerfiles in the images subdir
>    c) after b) merges, create a new PR here that deletes the dockerfiles in the top level dir
> 
> I'll let you decide what you want to do

I took the first option @gabemontero , I removed the images directory and just let it the dockerfiles in the top level dir

**Review by gabemontero — APPROVED** [review]
just to make sure, I checked out this PR and was able to make build-*-image all 3 images, as well as run `make test-e2e` and see the main_test.go change function correctly

/approve
/lgtm

**Comment by gabemontero** [comment]
/label px-approved

no end user impact

**Comment by gabemontero** [comment]
/label qe-approved

no qe impact yet, that will come with the real impl of the webhook

**Comment by gabemontero** [comment]
/label docs-approved

no doc impact yet, that will come with the real impl of the webhook

**Comment by gabemontero** [comment]
/lgtm cancel

missed something ... from the images job:

`error: error copying source path /go/src/github.com/openshift/csi-driver-shared-resource/images/. to .: exit status 1`

**Comment by gabemontero** [comment]
sigh .... we need to just undo the context dir stuff in https://github.com/openshift/release/pull/27887 before this will build clean @jkhelil

**Comment by gabemontero** [comment]
ok have https://github.com/openshift/release/pull/27998 ^^ up

**Comment by gabemontero** [comment]
ok we have clean images 

/lgtm

**Pull request merged**
